lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1916,6 +1916,11 @@ LogicalResult FIRRTLLowering::visitDecl(RegResetOp op) {
   };
 
   if (op.resetSignal().getType().isa<AsyncResetType>()) {
+    if (!firrtl::isConstant(op.resetValue()))
+      return op.emitError(
+                   "register with async reset requires constant reset value")
+                 .attachNote(op.resetValue().getLoc())
+             << "reset value defined here:";
```

Comment:
Can this just be a constraint on regreset in general rather than checked at lowering?

Reply:
That's what I had originally, but it gets hairy for the early parts of the pipeline, where the reset value might go through wires and subfield accesses in some convoluted fashion. In that case, you'd have to make sure that `isConstant` is adequately powerful to look through these things. In hindsight, that might actually be the right way forward: if we have a semantic checker pass, that could use the same IR query as the regreset verifier to ensure const-ness. Adding this to `LowerToHW` would at least allow us to catch the error in the current state of the code base, and the semantics checker / `isConstant` could then come in as a cleaner solution.

Reply:
I think this is a reasonable way to handle it - this is saying that it has to be constant by lowering time, but can be non-constant through the firrtl pipeline.  While that is problematic from an overall design perspective, it is consistent with SFC's approach, so I think we should follow suit.